server/services/status-history-logger.js

```diff
@@ -41,10 +43,29 @@ var storeLog = function (storage, logPath/*, options */) {
 	};
 };
 
+var saveRejectionReason = function (event) {
+	var status;
+	if (event.type !== 'direct') return;
+	if (!event.path.startsWith('processingSteps')) return;
+	status = unserializeValue(event.data.value);
+	if (status !== 'rejected' && status !== 'sentBack') return;
+
+	return mano.queryMemoryDb([event.ownerId], 'businessProcessRejectionReasons', {
```

Comment:
Let's require queryMemoryDb up front like we usally do in the beginning of the module. Also, no need to return anything as we consume with `done` (what you return here right now is `undefined` [as that is what is returned from done]).